### PR TITLE
Update instructions for making a release

### LINKED ISSUE #12: Conda-forge Bot automatically creates new PRs on new release
After noticing that conda-forge has a bot that automatically creates a new PR on the conda-forge feedstock repository after a new release of the project I think we should update the instructions on `MAINTENANCE.md`.
As I don't have much experience with conda-forge, would you @leouieda update it?
I don't exactly know how it's working and if the bot is bullet proof or could fail on some occasions.

### PR DESCRIPTION
**Pull request by santisoler**
Add instructions for updating release date on Zenodo entry.
Add instructions on how to fill the GitHub release information.
Add instructions for citing JOSS paper.
Fix typo and remove trailing spaces on `MAINTENANCE.md`.

Fixes #12

### CODE REVIEW

File: MAINTENANCE.md [hunk 1]
```diff
@@ -113,7 +114,12 @@ a **New upload** for it inside the
     ```
 
 6. Add a link to the new release version documentation in `README.rst`.
-7. Open a new PR with the updated changelog.
+7. Update DOI badge in `README.rst`.
+8. Update citation on `CITATION.rst`.
+8. Find any appearance of the old DOI and check if it needs to be updated or
+   the new one must be added. You can perform the search through
+   `git grep <OLD-DOI>`.
```
Comment (leouieda) [inline MAINTENANCE.md]: This is only necessary if we're including the version specific DOI in these places. For Verde and Pooch, we have the JOSS papers so this isn't necessary. Might be best to include this in a separate checklist for this particular case.

Comment (santisoler) [reply MAINTENANCE.md]: You're right!

**Comment by leouieda** [comment]
@santisoler I updated the instructions here after #13. Feel free to make any edits and merge after we have all the current bot PRs merged.

**Comment by santisoler** [comment]
This is looking awesome @leouieda! I'm really glad we took the discussion about releasing and authorship. I think we are having a much clearer workflow!
I think this is good to go!

**Pull request merged**